tests/integration_tests/test_applications_from_config.py

```diff
@@ -134,6 +135,42 @@ def compare_ecsv_files(file1, file2, tolerance=1.0e-5):
             assert np.allclose(table1[col_name], table2[col_name], rtol=tolerance)
 
 
+def assert_file_type(file_type, file_name):
+    """
+    Assert that the file is of the given type.
+
+    Parameters
+    ----------
+    file_type: str
+        File type (json, yaml).
+    file_name: str
+        File name.
+
+    """
+
+    if file_type == "json":
+        try:
+            with open(file_name, "r", encoding="utf-8") as file:
+                json.load(file)
+            return True
+        except (json.JSONDecodeError, FileNotFoundError):
+            return False
+    if file_type == "yaml" or file_type == "yml":
+        try:
+            with open(file_name, "r", encoding="utf-8") as file:
+                yaml.safe_load(file)
+            return True
+        except (yaml.YAMLError, FileNotFoundError):
+            return False
+
+    # no dedicated tests for other file types, checking suffix only
+    logger.info(f"File type test is checking suffix only for {file_name} (suffix: {file_type}))")
+    if file_name.suffix[1:] == file_type:
+        return True
```

Comment:
This is the redundant part I mentioned above. There is no need for this part, given that the existence of the output file (with suffix) is already checked by the other test.

Reply:
Not really - any file type which is not json or yaml is tested by this (too) simplistic test of comparing the file suffix. For yaml/json, these lines are not reached. 

I agree with you that as discussed above this should be replaced by explicit tests for each file type. Something to be added in future.

Reply:
I agree this is not redundant for yaml and json files, but for all the other output files in which we have both `OUTPUT_FILE` and `FILE_TYPE` tests, there will be a repetition. I agree that we can leave it like this for now because we only use `FILE_TYPE` to check json and yaml files, but something to keep in mind when expanding the function.